File: monitoring-and-billing-platform/src/main/java/com/water/monitoring_and_billing_platform/service/SupportTicketService.java
package com.water.monitoring_and_billing_platform.service;

import com.water.monitoring_and_billing_platform.dto.*;

import java.util.List;

public interface SupportTicketService {

    SupportTicketResponse createTicket(String userEmail, SupportTicketRequest request);

    List<SupportTicketResponse> getMySubmittedTickets(String userEmail);

    List<SupportTicketResponse> getCommunityInboxTickets(String userEmail);

    List<SupportTicketResponse> getMainAdminTickets(String userEmail);

    SupportTicketResponse getTicketDetails(String userEmail, Long ticketId);

    SupportTicketReplyResponse addReply(String userEmail, Long ticketId, SupportTicketReplyRequest request);

    List<SupportTicketReplyResponse> getTicketReplies(String userEmail, Long ticketId);

    SupportTicketResponse updateTicketStatus(String userEmail, Long ticketId, SupportTicketStatusUpdateRequest request);

    SupportTicketResponse closeTicket(String userEmail, Long ticketId);
}

File: frontend/src/services/SupportTicketService.js
import api from './api';

export const SupportTicketService = {
    createTicket: async (ticketData) => {
        const response = await api.post('/support-tickets', ticketData);
        return response.data;
    },

    getMySubmittedTickets: async () => {
        const response = await api.get('/support-tickets/my-tickets');
        return response.data;
    },

    getCommunityInboxTickets: async () => {
        const response = await api.get('/support-tickets/community-inbox');
        return response.data;
    },

    getMainAdminTickets: async () => {
        const response = await api.get('/support-tickets/main-admin-inbox');
        return response.data;
    },

    getTicketDetails: async (id) => {
        const response = await api.get(`/support-tickets/${id}`);
        return response.data;
    },

    addReply: async (id, replyData) => {
        const response = await api.post(`/support-tickets/${id}/replies`, replyData);
        return response.data;
    },

    getTicketReplies: async (id) => {
        const response = await api.get(`/support-tickets/${id}/replies`);
        return response.data;
    },

    updateTicketStatus: async (id, statusData) => {
        const response = await api.put(`/support-tickets/${id}/status`, statusData);
        return response.data;
    },

    closeTicket: async (id) => {
        const response = await api.put(`/support-tickets/${id}/close`);
        return response.data;
    }
};

export default SupportTicketService;
